engine/runtime/src/main/java/org/enso/interpreter/runtime/Module.java

```diff
@@ -53,18 +53,18 @@
 /** Represents a source module with a known location. */
 @ExportLibrary(InteropLibrary.class)
 public final class Module implements EnsoObject {
-  private ModuleScope scope;
   private ModuleSources sources;
+  private QualifiedName name;
+  private final ModuleScope.Builder scopeBuilder;
```

Comment:
Reference to builder - eh, I really cannot make any (static) judgements about such a design. Having a builder here doesn't make any logical sense to me.